File: PolishPizza/admin/src/service/orderService.ts
import { api as axiosInstance } from '../api/api';
import type { OrderRequest, OrderResponse } from '../types/order';

export const OrderService = {
    
    getAll: async (): Promise<OrderResponse[]> => {
        const response = await axiosInstance.get<OrderResponse[]>('/order/orders'); 
        return response.data;
    },

    getOne: async (id: string): Promise<OrderResponse | null> => {
        const response = await axiosInstance.get<OrderResponse>(`/order/orders/${id}`);
        return response.data;
    },
    
   update: async (id: string, data: any) => {
        const response = await axiosInstance.put(`/order/orders/${id}`, data);
        return response.data;
    },
    
    create: async (data: OrderRequest) => {
        if (data.type === 'DELIVERY') {
            const response = await axiosInstance.post(`/order/orders/delivery`, data);
            return response.data;
        } else {
            const response = await axiosInstance.post(`/order/orders`, data);
            return response.data;
        }
    },

    delete: async(id: string): Promise<void> => {
        const response = await axiosInstance.delete(`/order/orders/${id}`);
        return response.data;
    }
};

File: PolishPizza/order/src/main/java/pizzeria/orders/order/service/OrderService.java
package pizzeria.orders.order.service;

import org.springframework.data.domain.Sort;
import pizzeria.orders.order.dto.request.OrderDeliveryRequest;
import pizzeria.orders.order.dto.request.OrderPatchRequest;
import pizzeria.orders.order.dto.request.OrderRequest;
import pizzeria.orders.order.dto.response.OrderResponse;

import java.util.List;
import java.util.UUID;

public interface OrderService {
    List<OrderResponse> getAllOrders(UUID userId, String roles);

    OrderResponse getOrderById(UUID orderId, UUID userId, String roles);

    OrderResponse save(OrderRequest request, UUID userId);

    OrderResponse save(OrderDeliveryRequest request, UUID userId);

    void delete(UUID orderId, UUID userId);

    OrderResponse update(UUID orderId, OrderRequest request);

    OrderResponse patch(UUID orderId, OrderPatchRequest request);
}
